xsuite/src/cli/newWalletCmd.ts

```diff
@@ -37,19 +44,45 @@ const action = async ({
       return;
     }
   } else {
-    keystore = Keystore.createFile_unsafe(walletPath, password);
+    if (pemPath === undefined && oldWalletPath === undefined) {
+      keystore = Keystore.createFile_unsafe(walletPath, password);
+    } else {
+      if (pemPath !== undefined) {
+        const secretKey = UserSecretKey.fromPem(
+          fs.readFileSync(pemPath, { encoding: "utf8" }),
+        );
+        const wallet = UserWallet.fromSecretKey({ secretKey, password });
+        const data = wallet.toJSON();
+        keystore = Keystore.createFile_unsafe(walletPath, password, data);
+      } else if (oldWalletPath !== undefined) {
+        const oldKeystore = await Keystore.fromFile(oldWalletPath);
+        const oldKeystorePrivateKey = oldKeystore.getSecretKey();
```

Comment:
- This code would not work in case it is a mnemonic keystore? Ideally the code should be agnostic of the kind of the keystore.
- Adding a second test to test in case the keystore is a mnemonic would be great.